x-pack/test_serverless/api_integration/test_suites/observability/slos/delete_slo.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SO_SLO_TYPE } from '@kbn/observability-plugin/server/saved_objects';
+import { cleanup, generate } from '@kbn/infra-forge';
+import expect from '@kbn/expect';
+import { ALL_VALUE } from '@kbn/slo-schema';
+import {
+  getSLOSummaryTransformId,
+  getSLOTransformId,
+  getSLOSummaryPipelineId,
+} from '@kbn/observability-plugin/common/slo/constants';
+import {
+  SLO_DESTINATION_INDEX_PATTERN,
+  SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+} from '@kbn/observability-plugin/common/slo/constants';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+export default function ({ getService }: FtrProviderContext) {
+  const esClient = getService('es');
+  const logger = getService('log');
+  const kibanaServer = getService('kibanaServer');
+  const sloApi = getService('sloApi');
+  const transform = getService('transform');
+  const retry = getService('retry');
+
+  const esDeleteAllIndices = getService('esDeleteAllIndices');
+  const dataViewApi = getService('dataViewApi');
+
+  const fetchSloSummaryPipeline = async (
+    client: ElasticsearchClient,
+    sloId: string,
+    sloRevision: number
+  ) => {
+    try {
+      return await esClient.ingest.getPipeline({
+        id: getSLOSummaryPipelineId(sloId, sloRevision),
+      });
+    } catch (error) {
+      // The GET /_ingest/pipeline API returns an empty object on 404 Not Found. If there are no SLO
+      // pipelines then return an empty record of pipelines
+      return {};
+    }
+  };
+  describe('delete_slo', () => {
+    // DATE_VIEW should match the index template:
+    // x-pack/packages/kbn-infra-forge/src/data_sources/composable/template.json
+    const DATE_VIEW = 'kbn-data-forge-fake_hosts';
+    const DATA_VIEW_ID = 'data-view-id';
+    let infraDataIndex: string;
+    let sloId: string;
+
+    before(async () => {
+      await sloApi.deleteAllSLOs();
+
+      infraDataIndex = await generate({
+        esClient,
+        lookback: 'now-15m',
+        logger,
+      });
+      await dataViewApi.create({
+        name: DATE_VIEW,
+        id: DATA_VIEW_ID,
+        title: DATE_VIEW,
+      });
+      await kibanaServer.savedObjects.cleanStandardList();
+    });
+
+    after(async () => {
+      await dataViewApi.delete({
+        id: DATA_VIEW_ID,
+      });
+      await sloApi.deleteAllSLOs();
+      await esDeleteAllIndices([infraDataIndex]);
+      await cleanup({ esClient, logger });
+    });
+
+    describe('non partition by SLO', () => {
+      it('deletes the SLO definition, transforms, ingest pipeline and data', async () => {
```

Comment:
I was just wondering what is considered best practice regarding testing one thing at separate `it` statements versus testing many different things (SO definition, transforms etc) in one single `it` statement. 

Initially I had structured my [file differently](https://github.com/elastic/kibana/blob/35e95913d068a6708f460adada4dcc51bc0aad88/x-pack/test_serverless/api_integration/test_suites/observability/slos/delete_slo.ts), having multiple `it` statements. But I changed to a single test case, because 
- otherwise whole test suite needs more time to run. At every single test case it would need to create an SLO versus creating the SLO only once
- I felt it would be more flaky (? or maybe not, I am not sure) if not cleaning up properly before/after each test
- However I liked the structure more, when I had multiple it statements testing one thing at a time

Any feedback on this?

Reply:
If you want to group tests do it using a suite (or `describe` for this syntax). I think it's OK to have a suite where each test depends on the success of the previous test in the suite then clean everything up after each suite. Maybe add a comment before the `describe` that let's people know that we expect them to skip the entire suite and not just the individual test if there is an issue.